server/api/users.js

```diff
@@ -28,4 +36,43 @@ router.get('/users/:id', async (req, res, next) => {
   }
 });
 
+/* GET user by wallet. */
+// req with jwt: return db record
+// req without jwt / no db record: return challenge
+// no likecoin id: return 404
+router.get('/users/wallet/:wallet', jwtAuthNoBlock, async (req, res, next) => {
+  try {
+    const { wallet } = req.params;
+    if (req.user && req.user.wallet === wallet) {
+      const user = await sequelize.user.findOne(
+        { where: { wallet } },
+        { raw: true }
+      );
+      if (user) {
+        res.json(user);
+        return;
+      }
+    }
+    const payload = await getUserChallenge(wallet);
+    if (!payload || !payload.wallet) res.sendStatus(404);
+    res.json(payload);
+  } catch (err) {
+    next(err);
+  }
+});
+
+router.post('/users/login', async (req, res, next) => {
+  try {
+    const { wallet } = req.body;
+    const likecoinUser = await postUserChallenge(req.body);
+    const { likecoinId } = likecoinUser;
+    const token = jwtSign({ createdTs: Date.now(), likecoinId, wallet });
+    res.cookie('auth', token, AUTH_COOKIE_OPTION);
+    res.status(200).json(likecoinUser);
+    await sequelize.user.upsert(likecoinUser);
```

Comment:
`if `await sequelize.user.upsert(likecoinUser);` fails will cause session to be useless, then res.json should be sent after this is done